test/code/providers/runas_provider/runasprovider_test.cpp

```diff
@@ -59,6 +59,7 @@ class SCXRunAsProviderTest : public CPPUNIT_NS::TestFixture
     CPPUNIT_TEST( TestDoInvokeMethodScriptFailed );
     CPPUNIT_TEST( TestDoInvokeMethodScriptNonSH );
     CPPUNIT_TEST( TestDoInvokeMethodScriptNoHashBang );
+    CPPUNIT_TEST( TestDoInvokeMethodScriptTmpDir );
```

Comment:
Very nice that you have a unit test for this, excellent! You need PAL level tests for that functionality too, though.